components/tikv_kv/src/stats.rs

```diff
@@ -16,6 +19,63 @@ const STAT_NEXT_TOMBSTONE: &str = "next_tombstone";
 const STAT_PREV_TOMBSTONE: &str = "prev_tombstone";
 const STAT_SEEK_TOMBSTONE: &str = "seek_tombstone";
 const STAT_SEEK_FOR_PREV_TOMBSTONE: &str = "seek_for_prev_tombstone";
+const STAT_TTL_TOMBSTONE: &str = "ttl_tombstone";
+
+thread_local! {
+    pub static TTL_TOMBSTONE : RefCell<usize> = RefCell::new(0);
+}
+
+pub enum StatsKind {
+    Next,
+    Prev,
+    Seek,
+    SeekForPrev,
+}
+
+pub struct StatsCollector<'a> {
+    stats: &'a mut CfStatistics,
+    kind: StatsKind,
+
+    internal_tombstone: usize,
+    ttl_tombstone: usize,
+}
+
+impl<'a> StatsCollector<'a> {
+    pub fn new(kind: StatsKind, stats: &'a mut CfStatistics) -> Self {
+        StatsCollector {
+            stats,
+            kind,
+            internal_tombstone: PerfContext::get().internal_delete_skipped_count() as usize,
+            ttl_tombstone: TTL_TOMBSTONE.with(|m| *m.borrow()),
+        }
+    }
+}
+
+impl Drop for StatsCollector<'_> {
+    fn drop(&mut self) {
+        self.stats.ttl_tombstone += TTL_TOMBSTONE.with(|m| *m.borrow()) - self.ttl_tombstone;
```

Comment:
I'm afraid it's wrong. The future can be yield and fetched into other thread.

Reply:
In the span of StatsCollector lifetime, we just call rocksdb iter API. We don't call a `.await` in there, so the future would not be yield.